sound/soc/sof/amd/acp-loader.c

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+//
+// This file is provided under a dual BSD/GPLv2 license. When using or
+// redistributing this file, you may do so under either license.
+//
+// Copyright(c) 2021 Advanced Micro Devices, Inc.
+//
+// Authors: Ajit Kumar Pandey <AjitKumar.Pandey@amd.com>
+
+/*
+ * Hardware interface for ACP DSP Firmware binaries loader
+ */
+
+#include <linux/firmware.h>
+#include <linux/module.h>
+#include <linux/pci.h>
```

Comment:
I notice you removed multiple header includes from this and other files in this PR. Removing superfluous headers is good, but the required headers should be included. Quoting Documentation/process/submit-checklist.rst:
> 1) If you use a facility then #include the file that defines/declares
> that facility.  Don't depend on other header files pulling in ones
> that you use.

e.g. in this case you should explicitly include sof-priv.h, kernel.h, string.h, errno.h, device.h etc.

Reply:
Ok , we will do this in follow up PR with some fix up patch for headers.

Reply:
Is it really required to include all headers as mentioned in document ? because we don't see this been really followed in SOF core base and ASoC base. If we strictly need to follow that "quoted statement" we might need a fixup! patch for complete patch chain. For Eg: dev_dbg()/dev_err() facility is been used in all driver modules then **kernel.h** needs to be included in all c files where dev_dbg()/dev_err() has been used, but we don't see the same.

Reply:
strictly speaking that would be dev_printk.h, but AFAICS including device.h is also considered to be acceptable for this. The final word on this is certainly not mine, and yes, we're aware that this rule isn't *perfectly strictly* followed in SOF sources and even elsewhere around the kernel. Also, there are many cases where you actually *should* include a different file, not the one where the actual definition is made. dev_printk.h is one such example, but there are even stronger examples for that. E.g. if you need `ffs()` you don't `#include <asm/bitops/ffs.h>` you `#include <linux/bitops.h>` instead. But still you do include bitops.h instead of relying on it being included by kernel.h.

Reply:
@plbossart : Is it a mandatory at this moment? Do we really need a fix up for headers inclusion?

Reply:
@bhiregoudar it's strongly recommended. It's a difficult rule to apply and enforce, but the guidance is clear: try to avoid including headers in other headers, and only include them when they are used.

In many cases going through the list of headers actually has the effect to removing things that are not necessary and are included for no good reason.

Reply:
Sorry for delay in response. We won't be able to take these suggested changes as a fix up patch, but will create a separate new PR to accommodate these changes later

Reply:
please create a new PR with commits starting by 'fixup! ' so they can be auto-squashed.